scripts/quests/bastok/A_Foremans_Best_Friend.lua

```diff
@@ -56,8 +54,11 @@ quest.sections =
             onEventFinish =
             {
                 [112] = function(player, csid, option, npc)
-                    if quest:complete(player) then
-                        player:confirmTrade()
+                    player:confirmTrade()
+                    if player:hasKeyItem(xi.ki.MAP_OF_THE_GUSGEN_MINES) then
+                        player:addExp(2000)
+                    else
+                        npcUtil.giveKeyItem(player, xi.ki.MAP_OF_THE_GUSGEN_MINES)
```

Comment:
We still want the quest:complete() check, I would think

Reply:
Solid point.

Reply:
Tis fixed my friend.